subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -159,6 +169,32 @@ public void configureUserAgent(HttpClientBuilder builder) {
         builder.setUserAgent(UriTextResource.getUserAgentString());
     }
 
+    private void configureCookieSpecRegistry(HttpClientBuilder builder) {
+        PublicSuffixMatcher publicSuffixMatcher = PublicSuffixMatcherLoader.getDefault();
```

Comment:
I think we could avoid a lot of this reimplementation of HttpClient code by just calling `CookieSpecRegistries.createDefault` and then using `register()` to replace the ones we need to.